arbstate/inbox.go

```diff
@@ -165,6 +166,55 @@ func parseSequencerMessage(ctx context.Context, batchNum uint64, batchBlockHash
 	return parsedMsg, nil
 }
 
+func RecoverPayloadFromAvailBatch(ctx context.Context, batchNum uint64, sequencerMsg []byte, availDAReader avail.DataAvailabilityReader, preimages map[arbutil.PreimageType]map[common.Hash][]byte) ([]byte, error) {
```

Comment:
can we reduce the number of arguments and split using few internal functions

Reply:
No, it can't be changed, `RecoverPayloadfromAvailBatch` is inheriting the #2155 
`DataAvailabilityProvider` interface:- https://github.com/OffchainLabs/nitro/pull/2155/files#diff-e24cacc680b8e01e6610107edccd3114f681d583376a9ffc9614fa87df314b32R307